scripts/sysdig-probe-loader

```diff
@@ -192,12 +192,24 @@ load_kernel_probe() {
 	URL=$(echo "${SYSDIG_PROBE_URL}/${SYSDIG_REPOSITORY}/sysdig-probe-binaries/${SYSDIG_PROBE_FILENAME}" | sed s/+/%2B/g)
 
 	echo "* Trying to download precompiled module from ${URL}"
-	if curl --create-dirs "${SYSDIG_PROBE_CURL_OPTIONS}" -o "${HOME}/.sysdig/${SYSDIG_PROBE_FILENAME}" "${URL}"; then
+
+	curl_out=$(curl --create-dirs "${SYSDIG_PROBE_CURL_OPTIONS}" -o "${HOME}/.sysdig/${SYSDIG_PROBE_FILENAME}" "${URL}" 2>&1)
+	if [ "$?" = "0" ]; then
 		echo "Download succeeded, loading module"
 		insmod "${HOME}/.sysdig/${SYSDIG_PROBE_FILENAME}"
 		exit $?
 	else
-		echo "Download failed, consider compiling your own ${PROBE_NAME} and loading it or getting in touch with the sysdig community"
+		if [[ "$curl_out" =~ "404 Not Found" ]]; then
```

Comment:
This change slightly alters the behavior of the script, as the helpful error message is only printed when `curl` prints a 404 error. 

However, I don't rightly understand how the original script even works, since 1curl` shouldn't return a nonzero exit code for a 404 error unless it's given the `-f` flag. I'm a little reluctant to change the behavior without understanding the original, but maybe you do understand it and if so then go for it :)

Reply:
I tested this bit of code. The `curl` line threw a error status of 2 when the 404 error is emitted. The previous code also relied on this error status being 0 to continue or exit if not. This code change does not alter code behaviour. Only improves error messages.